src/tests/java/picard/sam/markduplicates/SimpleMarkDuplicatesWithMateCigar.java

```diff
@@ -0,0 +1,211 @@
+        
+        iterator.setScoringStrategy(this.DUPLICATE_SCORING_STRATEGY);
+
+        // progress logger!
+        final ProgressLogger progress = new ProgressLogger(log, (int) 1e6, "Read");
+
+        int numDuplicates = 0;
+        
+        libraryIdGenerator = new LibraryIdGenerator(headerAndIterator.header);
+        
+        for (final DuplicateSet duplicateSet : new IterableAdapter<DuplicateSet>(iterator)) {
+            final SAMRecord representative = duplicateSet.getRepresentative();
+            final boolean doOpticalDuplicateTracking = representative.getReadPairedFlag() &&
+                    representative.getFirstOfPairFlag() &&
+                    !representative.getReadUnmappedFlag() &&
+                    !representative.getMateUnmappedFlag();
+            final Set<String> readEndsSeen = new HashSet<String>();
+            
+            final List<ReadEnds> readEnds = new ArrayList<ReadEnds>();
+            for (final SAMRecord record : duplicateSet.getRecords()) {
+
+                if (!record.isSecondaryOrSupplementary()) {
+                    final String library = LibraryIdGenerator.getLibraryName(header, record);
+                    DuplicationMetrics metrics = libraryIdGenerator.getMetricsByLibrary(library);
+                    if (metrics == null) {
+                        metrics = new DuplicationMetrics();
+                        metrics.LIBRARY = library;
+                        libraryIdGenerator.addMetricsByLibrary(library, metrics);
+                    }
+
+                    // First bring the simple metrics up to date
+                    if (record.getReadUnmappedFlag()) {
+                        ++metrics.UNMAPPED_READS;
+                    } else if (!record.getReadPairedFlag() || record.getMateUnmappedFlag()) {
+                        ++metrics.UNPAIRED_READS_EXAMINED;
+                    } else {
+                        ++metrics.READ_PAIRS_EXAMINED; // will need to be divided by 2 at the end
+                    }
+
+                    if (record.getDuplicateReadFlag()) {
+                        // Update the duplication metrics
+                        if (!record.getReadPairedFlag() || record.getMateUnmappedFlag()) {
+                            ++metrics.UNPAIRED_READ_DUPLICATES;
+                        } else {
+                            ++metrics.READ_PAIR_DUPLICATES;// will need to be divided by 2 at the end
+                        }
+                        numDuplicates++;
+                    }
+                    
+                    // To track optical duplicates, store a set of locations for mapped pairs, first end only
+                    if (doOpticalDuplicateTracking &&
+                            record.getReadPairedFlag() &&
+                            !record.getReadUnmappedFlag() &&
+                            !record.getMateUnmappedFlag() &&
+                            !readEndsSeen.contains(record.getReadName())) {
```

Comment:
We also care that the representative's mate is also mapped (not unmapped).  

I am assuming you mean ```duplicate pairs could be in either order`` to be in terms of orientation.  For PCR duplicates yes, for optical duplicates no.